com.unity.netcode.gameobjects/Runtime/NetworkVariable/CollectionSerializationUtility.cs

```diff
@@ -0,0 +1,664 @@
+using System;
+using System.Collections.Generic;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Mathematics;
+
+namespace Unity.Netcode
+{
+    internal static class CollectionSerializationUtility
+    {
+        public static void WriteNativeArrayDelta<T>(FastBufferWriter writer, ref NativeArray<T> value, ref NativeArray<T> previousValue) where T : unmanaged
+        {
+            using var changes = new ResizableBitVector(Allocator.Temp);
+            int minLength = math.min(value.Length, previousValue.Length);
+            var numChanges = 0;
+            for (var i = 0; i < minLength; ++i)
+            {
+                var val = value[i];
+                var prevVal = previousValue[i];
+                if (!NetworkVariableSerialization<T>.AreEqual(ref val, ref prevVal))
+                {
+                    ++numChanges;
+                    changes.Set(i);
+                }
+            }
+
+            for (var i = previousValue.Length; i < value.Length; ++i)
+            {
+                ++numChanges;
+                changes.Set(i);
+            }
+
+            if (changes.GetSerializedSize() + FastBufferWriter.GetWriteSize<T>() * numChanges > FastBufferWriter.GetWriteSize<T>() * value.Length)
+            {
+                writer.WriteByteSafe(1);
+                writer.WriteValueSafe(value);
+                return;
+            }
+            writer.WriteByte(0);
```

Comment:
One example of comments would be the byte written here (and above) and what it means?
It would help down the road without having to re-synch with everything happening in here.